src/HealthChecks.Rabbitmq/DependencyInjection/RabbitMQHealthCheckBuilderExtensions.cs

```diff
@@ -153,6 +157,10 @@ public static class RabbitMQHealthCheckBuilderExtensions
     /// <summary>
     /// Add a health check for RabbitMQ services using <see cref="IConnectionFactory"/> factory function.
     /// </summary>
+    /// <remarks>
+    /// <paramref name="connectionFactoryFactory"/> will be called each time the healthcheck route is requested. However
+    /// the created <see cref="IConnection"/> will be reused.
```

Comment:
>      /// the created <see cref="IConnection"/> will be reused.

not true, options object is created each time so `GetOrAdd` falls into delegate creting connection

Reply:
To use approach with options as a dictionary key you should move options creation in all methods out of  HealthCheckRegistration's delegate.